File: platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/annotations/CoreElementProvider.java
package org.maplibre.android.annotations;

import androidx.annotation.Nullable;

import org.maplibre.android.style.layers.Layer;
import org.maplibre.android.style.sources.GeoJsonOptions;
import org.maplibre.android.style.sources.GeoJsonSource;

interface CoreElementProvider<L extends Layer> {

    String getLayerId();

    String getSourceId();

    L getLayer();

    GeoJsonSource getSource(@Nullable GeoJsonOptions geoJsonOptions);
}

File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/annotations/CoreElementProvider.kt
package org.maplibre.android.annotations

import org.maplibre.android.style.layers.Layer
import org.maplibre.android.style.sources.GeoJsonOptions
import org.maplibre.android.style.sources.GeoJsonSource

interface CoreElementProvider<L : Layer> {
    val layerId: String
    val sourceId: String
    val layer: L
    fun getSource(geoJsonOptions: GeoJsonOptions?): GeoJsonSource
}
